go/exec/exec.go

```diff
@@ -193,6 +195,23 @@ func (e *Exec) Prepare() error {
 	return nil
 }
 
+// ExecuteWithTimeout will call execution's Execute method with the given timeout.
+// Note that a timeout too small will result in
+func (e Exec) ExecuteWithTimeout(timeout time.Duration) error {
+	errs := make(chan error)
+
+	go func() {
+		errs <- e.Execute()
+	}()
+
+	select {
+	case err := <-errs:
+		return err
+	case <-time.After(timeout):
+		return errors.New(ErrorExecutionTimeout)
```

Comment:
should we not cleanup e over here. Even if it timedout, the server needs to be closed

Reply:
We let the caller of `ExecuteWithTimeout` deal with the error, as we do with `Execute`